src/main/java/io/github/retrooper/packetevents/event/manager/EventManagerModern.java

```diff
@@ -105,6 +105,32 @@ public synchronized void registerListeners(PacketListenerAbstract... listeners)
         }
     }
 
+    /**
+     * Unregister the dynamic packet event listener.
+     *
+     * @param listener {@link PacketListenerAbstract}
+     */
+    public synchronized void unregisterListener(final PacketListenerAbstract listener) {
+        byte priority = listener.getPriority().getId();
+        HashSet<PacketListenerAbstract> listenerSet = listenersMap.get(priority);
+        if (listenerSet == null) {
+            return;
+        }
+        listenerSet.remove(listener);
+        listenersMap.put(priority, listenerSet);
```

Comment:
This line is unnecessary, this priority already points to this hashset

Reply:
I just copied what the registration method did. I'll change this shortly.